pallets/vested-rewards/src/lib.rs

```diff
@@ -120,6 +123,210 @@ pub struct CrowdloanUserInfo<AssetId> {
 pub use weights::WeightInfo;
 
 impl<T: Config> Pallet<T> {
+    #[cfg(feature = "wip")] // ORML multi asset vesting
+    fn do_claim_unlocked(
+        asset_id: AssetIdOf<T>,
+        who: &T::AccountId,
+    ) -> Result<BalanceOf<T>, DispatchError> {
+        let (locked, other_count): (Balance, u32) = Self::locked_balance(asset_id, who)?;
+        if locked.is_zero() && other_count.is_zero() {
+            // cleanup the storage and unlock the fund
+            <VestingSchedules<T>>::remove(who);
+            T::Currency::remove_lock(VESTING_LOCK_ID, asset_id, who)?;
+        } else if locked.is_zero() {
+            T::Currency::remove_lock(VESTING_LOCK_ID, asset_id, who)?;
+        } else {
+            T::Currency::set_lock(VESTING_LOCK_ID, asset_id, who, locked)?;
+        }
+        Ok(locked)
+    }
+
+    #[cfg(feature = "wip")] // ORML multi asset vesting
+    /// Returns locked balance based on current block number.
+    fn locked_balance(
+        asset_id: AssetIdOf<T>,
+        who: &T::AccountId,
+    ) -> Result<(BalanceOf<T>, u32), DispatchError> {
+        let now = frame_system::Pallet::<T>::current_block_number();
+        <VestingSchedules<T>>::mutate_exists(who, |maybe_schedules| {
+            // Calculated to delete if there are no locks
+            let mut other_count: u32 = 0;
+            let mut total_asset: BalanceOf<T> = Zero::zero();
+            if let Some(schedules) = maybe_schedules.as_mut() {
+                let mut valid_schedules = Vec::new();
+
+                for s in schedules.iter() {
+                    match s.locked_amount(now) {
+                        Some(amount) if s.asset_id() == asset_id => {
+                            total_asset = total_asset.saturating_add(amount);
+                            if !amount.is_zero() {
+                                valid_schedules.push(s.clone());
+                            }
+                        }
+                        Some(_) => {
+                            // Keep schedules for other assets
+                            other_count += 1;
+                            valid_schedules.push(s.clone());
+                        }
+                        None => {
+                            return Err(Error::<T>::ArithmeticError.into());
+                        }
+                    }
+                }
+
+                *maybe_schedules =
+                    if valid_schedules.is_empty() {
+                        None
+                    } else {
+                        Some(BoundedVec::try_from(valid_schedules).map_err(|_| {
+                            DispatchError::Other("Error while convert to BoundedVec")
+                        })?)
+                    };
+            }
```

Comment:
Reduce indentation and hopefully improve readability
```suggestion
            let Some(schedules) = maybe_schedules.as_mut() {
                return Ok((total_asset, other_count));
            }
            let mut valid_schedules = BoundedVec::new();
            for s in schedules.iter() {
                if s.asset_id() == asset_id {
                    let amount = s.locked_amount(now).ok_or(Error::<T>::ArithmeticError)?;
                    total_asset = total_asset.saturating_add(amount);
                    if !amount.is_zero() {
                        valid_schedules.try_push(s.clone()).ok_or(Error::<T>::MaxVestingSchedulesExceeded)?;
                    }
                } else {
                    // Keep schedules for other assets
                    other_count += 1;
                    valid_schedules.try_push(s.clone()).ok_or(Error::<T>::MaxVestingSchedulesExceeded)?;
                }
            }
```